### ISSUE TITLE
instagraal-polish

### ISSUE THREAD

**Issue by Overcraft90** [koszullab/instaGRAAL#36]
Hi @nadegeguiglielmoni,

I closed the previous issue as I solved my problems with the tool, specifically those related to scaffolding. I've been able to run it in local, and I got something after about 2-3 days (as you predicted) although the algorithm converged much earlier than expected.

After about 8-9 cycles I noted a sharp reduction in the _log_likelihood_ file, but following your advice I let `instaGRAAL` run for few more cycles just to make sure it was stable. This is what I got after 20 cycles:

![list_likelihood](https://user-images.githubusercontent.com/98895614/172059560-238606db-7b23-48e4-b710-2bc6b7287958.png)

Now, I was reading the section about curation. In particular, I would like to run `instagraal-polish -m polishing -i info_frags.txt -f contigs.fasta -o curated_assembly.fa`; however, for my understanding I would like to know exactly what to input after the _**-f**_ and the _**-o**_ flags. I think after the _**-f**_ I have to indicate the original FASTA genome, whereas after the _**-o**_ the output of `instaGRAAL` after the 20 cycles I run. Let me know, thanks.

**Comment by Overcraft90** [comment]
Hi there,

Just wondering whether you think the _list-likelihood_ converged but also, and mainly, what are the exact input after the _**-f**_ and the _**-o**_ flags. Thanks.

**Comment by nadegeguiglielmoni** [comment]
Hello,

Yes it looks like it has converged. For instagraal-polish, -f is the initial assembly that you scaffolded, and -o is the name you want to give to your scaffolded curated assembly. You should also use the parameter -j to specify the gap sequence (like NNNNNNNNNN). Then you can do gap filling to further improve the quality of your scaffolds.

**Comment by Overcraft90** [comment]
Hi,

Thanks I will do as you said and let you know. Out of curiosity, how long is going to take roughly for the `instagraal-polish`? Thanks again!

**Comment by Overcraft90** [comment]
Hi @nadegeguiglielmoni,

Last week I went on and tried `instagraal-polish`. The overall execution was pretty fast, maybe something around 5' which is great.

I think now that I experimented a bit with the tool, I would like to know how to open the _**.fasta**_ file generated after scaffolding with the `PretextView tool` so that I can compare the before and after.

Thanks in advance!

**Comment by Overcraft90** [comment]
Hi @nadegeguiglielmoni,

I'm proceeding as you recommended me. However, what is a good binning for a human haplotype? I'm just wondering if the 10kb indicated in the `hicstuff` example could work for me. Thanks in advance!

**Comment by Overcraft90** [comment]
Hi again a quick update,

Both `instaGRAAL` and `instagraal-polish` worked wonderfully. Following the Hi-C contact map visualized with `PertextView`:

BEFORE
![Screenshot from 2022-06-30 12-27-55](https://user-images.githubusercontent.com/98895614/176682484-0d44e1cc-4c48-4b65-b92b-2ab7af533610.png)
AFTER
![Screenshot from 2022-06-30 12-27-14](https://user-images.githubusercontent.com/98895614/176682533-7acc7f07-b059-4fad-836d-440faf9e52de.png)

Now, the only issue is that I had to resort to the arima_mapping_pipeline to get them; in fact, to one generated by `hicstuff` are very coarse and don't look that good, unfortunately... here an example:

BEFORE (with `hicstuff view`)
![contact_map_HG00733_hap1](https://user-images.githubusercontent.com/98895614/176683509-d2755c46-3e97-4643-bb2d-5644f65a14b9.png)

Again, I'm sure tweaking with parameters (_e.g._ bin size and other relevant ones) could improve things, but due to time constraints I had to go fro the ARIMA approach.

Well, I think this concludes my test on scaffolding. I believe `instaGRAAL` is great! I will be presenting a poster including the former results at a conference in Swiss with a main focus on Genome Graphs and Pangenomic.

Thanks again,
Matteo
